### ISSUE TITLE
Review Autocomplete navigation behavior

### ISSUE THREAD

**Issue by matuzalemsteles** [liferay/clay#5203]
We recently made some keyboard navigation behavior changes in DropDown which is now following the recommendations of w3c #5193, and autocomplete also had accessibility improvements along with a new composition #5108 the difference is that this improvement came in before we decided on the behavior keyboard navigation for DropDown which means that now we also need to change the keyboard navigation behavior of Autocomplete to just arrow keys and not allow tabs inside the Menu.

Now with the entry of #5193 and the behavior decision we can go with it and apply the keyboard navigation that was missing for Autocomplete following the [w3c recommendations](https://w3c.github.io/aria-practices/examples/combobox/combobox-autocomplete-list.html) with virtual focus instead of moving focus.

We also need this to be able to apply the combobox pattern in MultiSelect (#5194), since MultiSelect uses Autocomplete but using its old version, so it affects MultiSelect too, we will need to apply the collection pattern, this will be done when using the new Autocomplete behavior in MultiSelect.

The main changes we should make to the component with this issue are:

- Only allow keyboard navigation using arrow keys
- Navigation using virtual focus
- Add loop navigation
- Adding shortcuts like Home, End, pressing arrow down move the virtual focus to the last element, this should be compatible with virtualization, so we will need to implement more functionality in the `useNavigation` hook that is dealing with this.
